crates/bevy_ui/src/widget/text.rs

```diff
@@ -49,15 +64,26 @@ pub fn text_system(
     mut text_queries: ParamSet<(
         Query<Entity, Or<(Changed<Text>, Changed<Style>)>>,
         Query<Entity, (With<Text>, With<Style>)>,
-        Query<(&Text, &Style, &mut CalculatedSize)>,
+        Query<(&Text, &Style, &mut CalculatedSize, Option<&Parent>)>,
     )>,
+    node_query: Query<&Node>,
 ) {
-    let scale_factor = windows.scale_factor(WindowId::primary());
+    // FIXME: this will be problematic for multi-window UI's since its only checking the primary display
+    let (scale_factor, window_width_constraint, window_height_constraint) =
+        if let Some(window) = windows.get_primary() {
+            (window.scale_factor(), window.width(), window.height())
+        } else {
+            (1., f32::MAX, f32::MAX)
```

Comment:
this case seems like it shouldn't happen as it can't give correct values. maybe ignore / panic instead of running values without meaning

Reply:
Yeah I agree, I'll fix this when I get a chance.